lib/logger/lib/logger/error_handler.ex

```diff
@@ -94,22 +111,30 @@ defmodule Logger.ErrorHandler do
   defp ensure_pid(pid) when is_pid(pid), do: pid
   defp ensure_pid(_), do: self()
 
-  defp check_threshold(%{last_time: last_time, last_length: last_length,
-                         dropped: dropped, threshold: threshold} = state) do
-    {m, s, _} = current_time = :os.timestamp
+  defp check_threshold(
+         %{last_time: last_time, last_length: last_length, dropped: dropped, threshold: threshold} =
```

Comment:
Please remove the pattern matching out of the definition and into the function body.